.travis.yml

```diff
@@ -41,6 +41,7 @@ matrix:
         - cd dex-contracts && truffle compile && cd -
         - docker-compose up -d ganache-cli driver listener | while read line ; do echo "$(date)| $line"; done; # prints timestamps for profiling
       script:
+        - set -e # make this script fail as soon as any individual command fail
         - sh ./test/restart-containers.sh
```

Comment:
Unrelated, but do we need to restart the container before the very first test?

Reply:
Yes, because the restart script has the `truffle migrate` in it.

Reply:
Can we make this part of the before script? It feels a bit 🤡 to start a container just to restart it right afterwards.

Reply:
Ya, sure... I will change it here and see how you like it.

Reply:
Also, we could consider taking the truffle migrate out of the restart. here are a few options.

A: I think this looks stupid
```
      install:
        - git submodule update --init
        - npm install -C dex-contracts
        # driver/listener require compiled contracts
        - cd dex-contracts && truffle compile && cd -
        - docker-compose up -d ganache-cli driver listener | while read line ; do echo "$(date)| $line"; done; # prints timestamps for profiling
*       - sh ./test/restart-containers.sh
      script:
        - set -e # make this script fail as soon as any individual command fail
*
        - sh ./test/e2e-tests-deposit-withdraw.sh
        - sh ./test/restart-containers.sh
        - sh ./test/e2e-tests-auction.sh
        - sh ./test/restart-containers.sh
        - sh ./test/e2e-tests-standing-order.sh
```


B: This seems like a reasonable option
```
      install:
        - git submodule update --init
        - npm install -C dex-contracts
        # driver/listener require compiled contracts
        - cd dex-contracts && truffle compile && cd -
        - docker-compose up -d ganache-cli driver listener | while read line ; do echo "$(date)| $line"; done; # prints timestamps for profiling
      script:
        - set -e # make this script fail as soon as any individual command fail
*       - cd dex-contracts && retry truffle migrate && cd -
        - sh ./test/e2e-tests-deposit-withdraw.sh
        - sh ./test/restart-containers.sh
        - sh ./test/e2e-tests-auction.sh
        - sh ./test/restart-containers.sh
        - sh ./test/e2e-tests-standing-order.sh
```



C: We remove the last line from `restart-containers` (which is `cd dex-contracts && retry truffle migrate && cd -`) and do the following:
```
      install:
        - git submodule update --init
        - npm install -C dex-contracts
        # driver/listener require compiled contracts
        - cd dex-contracts && truffle compile && cd -
        - docker-compose up -d ganache-cli driver listener | while read line ; do echo "$(date)| $line"; done; # prints timestamps for profiling
      script:
        - set -e # make this script fail as soon as any individual command fail
*       - cd dex-contracts && retry truffle migrate && cd -
        - sh ./test/e2e-tests-deposit-withdraw.sh
        - sh ./test/restart-containers.sh
*       - cd dex-contracts && retry truffle migrate && cd -
        - sh ./test/e2e-tests-auction.sh
        - sh ./test/restart-containers.sh
*       - cd dex-contracts && retry truffle migrate && cd -
        - sh ./test/e2e-tests-standing-order.sh
```

I think option C ruins the purpose of making the tests independent.

Reply:
I like B the best

Reply:
Have decided to go with a variant of option `B`